### DISCUSSION TITLE
Modal prompt from Tabler

### DISCUSSION THREAD

**Discussion by ionv06** [sqlpage/SQLPage#570] [Q&A]
Hello,
I have a table with a list of records, I would like to put a delete button on each row, and the button to call a modal like the one in Tabler. How could this be achieved? I made a component for the modal, another one for the button (I couldn't use the existing button because it seems that it must have the property data-bs-toggle="modal" and data-bs-target="idModal"), but I can't display the button on each record in the table.
![image](https://github.com/user-attachments/assets/3bc2978a-9483-417e-bfaa-75a9e3381c46)

**Comment by lovasoa** [comment]
Hello !

  The easiest "native" way to implement delete confirmation is to use the alert component on a separate deletion page. See [the todo application example](https://github.com/lovasoa/SQLpage/blob/main/examples/todo%20application/) and [`delete.sql`](https://github.com/lovasoa/SQLpage/blob/main/examples/todo%20application/delete.sql) in particular.

If you are ready to create your own component, then you can include a script in it that will dynamically update your table after the fact to insert a modal button in each row.

The whole goal of SQLPage is to free you from thinking about these things, to implement your core functionality faster, though. I would just go with a traditional link that loads a new page with a confirmation message in an alert component.

**Comment by ionv06** [comment]
Thank you for your reply.  I'm already testing the version with the 'alert' component, it's fine, I'm just trying to find optimization solutions. My concern regarding this option is that if I have several hundred records displayed in the table, and to delete one I have to redirect back to the page, make a new 'select' and generate the table again, I would load the server too much. If there were more users who would simultaneously perform operations like this, the load would be even higher. All these are just ideas, questions I ask myself, I don't face them yet in production (at the moment there is no 'production' either ). I'm just trying to see how things work and find the best options

**Reply by lovasoa** [reply]
I encourage you to test with your data, but from experience, hundreds of records is generally not a problem, either for sqlpage of for a database, even under high load.

If you have millions of records, then indeed, it's better to avoid loading them all at once. In this case I would recommend pagination, or slicing the data based on one of its dimensions.

What kind of data are you working with, and what are you building ?

**Reply by ionv06** [reply]
I maintain an accounting software, something like an ERP. It is developed in visual fox pro - I know, it's old :) - and the databases are on MariaDb. I'm thinking of making some user interfaces with SQLPage so I'm evaluating, learning to use it.. I said hundreds of records above, but usually it's thousands or tens of thousands (accounting knows how to generate tens of records from a document). By the way.. how could I change the current database on the fly? On the same server I have several databases, with identical structure, each one represents a company and the user must be able to work in whichever company he wants. Can I use 'use X'?

**Reply by lovasoa** [reply]
Hey ! Currently SQLPage only supports connecting to a single database server at a time. In mariadb, if your user has access to multiple databases, you can select from them with `select DB1.my_table; select DB2.my_table; `. But if you need to dynamically select the database based on a query parameter, that is currently not possible. The way I would do it is with multiple sqlpage instances running over the same sql pages.

Something like this:

```sh
DATABASE_URL=mysql://localhost/company1 PORT=8000 ./sqlpage
DATABASE_URL=mysql://localhost/company2 PORT=8001 ./sqlpage
```

and an nginx to route requests to them: 

```nginx
map $host $db_port {
    company1.example.com 8000;
    company2.example.com 8001;
}

server {
    listen 80;
    server_name company1.example.com company2.example.com;

    location / {
        proxy_pass http://localhost:$db_port;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

```

**Reply by ionv06** [reply]
Thanks a lot for the idea. It's not exactly what I wanted, but it's a good starting point. I tested and it works with several instances of sqlpage (without ngnix at the moment)

**Reply by ionv06** [reply]
Hi, Ophir! I exchanged some messages with Nick Antonnacio recently about the possibility of changing the database dynamically. I told him that all I need is for a user to be able to dynamically change the database they are working on. For example, I have a mysql server with databases 'db1', 'db2', ... 'db99' - all with the same structure (they contain the same tables, the same structure). From the application created with SQLPage, the user must be able to change the active database 'with one click'. In other development environments, I use a 'use db2', after which I send the SQL instructions, without having to prefix the names of the database tables ('select id, document, value from documents', not '...from db2.documents')
Nick told me that it would be possible for you to implement such a thing, so I would be grateful if you would do that and it would certainly be useful to others.

**Reply by lovasoa** [reply]
Hi @ionv06 ! Do you have an email address where we can join you to discuss that ?

**Reply by heejit** [reply]
or if possible to use form variable as identifier (table_name or schema_name) in sql query.

select id,name from **:current_db**.item_master

**Reply by ionv06** [reply]
@lovasoa I wrote to you yesterday at contact[at]ophir.dev
